onnxruntime/python/tools/transformers/onnx_model.py

```diff
@@ -20,6 +21,13 @@ class OnnxModel:
     def __init__(self, model):
         self.model = model
         self.node_name_counter = {}
+        self.shape_infer_helper = SymbolicShapeInferenceHelper(self.model)
+
+    def infer_runtime_shape(self):
```

Comment:
How about updating API like:
```
   def infer_runtime_shape(self, dynamic_axis_mapping={}):
         helper = SymbolicShapeInferenceHelper(self.model)
         if helper.infer(dynamic_axis_mapping):
            return helper
         return None
```

In this way, user could decide whether to cache the shape inference result or re-run shape inference when graph is updated.